test/e2e/e2e_test.go

```diff
@@ -0,0 +1,286 @@
+// Package e2e_test implements simple integration test against a local cluster, whose config is loaded from the kubeconfig file.
+package e2e_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	sdkEvent "github.com/newrelic/infra-integrations-sdk/data/event"
+	"github.com/newrelic/nri-kube-events/pkg/events"
+	"github.com/newrelic/nri-kube-events/test/e2e"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// We must have a global MockedAgentSink because the infrastructure-sdk attempts to register global flags when the
+// agent sink is created, which results in a panic if multiple instantiations are attempted.
+var globalMockedAgentSink *e2e.MockedAgentSink
+
+// TestPodCreation checks that events related to pod creation are received.
+func TestPodCreation(t *testing.T) {
+	client, agentMock := initialize(t)
+
+	t.Log("Creating test namespace...")
+	ns, err := client.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "e2e-" + strings.ToLower(t.Name()) + "-",
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		t.Logf("could not create %s namespace: %v", ns, err)
```

Comment:
Test should fail if test namespace cannot be created.
```suggestion
		t.Fatalf("could not create %s namespace: %v", ns, err)
```